crates/cargo-wdk/src/actions/package/mod.rs

```diff
@@ -0,0 +1,370 @@
+            .canonicalize_path(cargo_metadata.workspace_root.clone().as_std_path())?;
+        if workspace_root.eq(working_dir) {
+            debug!("Running from workspace root");
+            let target_directory: PathBuf = target_directory.into();
+            for package in workspace_packages {
+                let package_root_path: PathBuf = package
+                    .manifest_path
+                    .parent()
+                    .expect("Unable to find package path from Cargo manifest path")
+                    .into();
+
+                let package_root_path = self
+                    .fs_provider
+                    .canonicalize_path(package_root_path.as_path())?;
+                debug!(
+                    "Processing workspace driver package: {}",
+                    package_root_path.display()
+                );
+                self.build_and_package(
+                    &package_root_path,
+                    &wdk_metadata,
+                    package,
+                    package.name.clone(),
+                    &target_directory,
+                )?;
+            }
+            return Ok(());
+        }
+        info!("Running from workspace member directory");
+        let package = workspace_packages.iter().find(|p| {
+            let package_root_path: PathBuf = p
+                .manifest_path
+                .parent()
+                .expect("Unable to find package path from Cargo manifest path")
+                .into();
+            self.fs_provider
+                .canonicalize_path(package_root_path.as_path())
+                .is_ok_and(|package_root_path| {
+                    debug!(
+                        "Processing workspace driver package: {}",
+                        package_root_path.display()
+                    );
+                    package_root_path.eq(working_dir)
+                })
+        });
+
+        if package.is_none() {
+            return Err(PackageProjectError::NotAWorkspaceMember(
+                working_dir.clone(),
+            ));
+        }
+
+        let package = package.expect("Package cannot be empty");
+        self.build_and_package(
+            working_dir,
+            &wdk_metadata,
+            package,
+            package.name.clone(),
+            target_directory.as_std_path(),
+        )?;
+
+        info!("Building and packaging completed successfully");
+
+        Ok(())
+    }
+
+    fn get_cargo_metadata(
+        &self,
+        working_dir: &Path,
+    ) -> Result<cargo_metadata::Metadata, PackageProjectError> {
+        let working_dir_path_trimmed: PathBuf = working_dir
+            .to_string_lossy()
+            .trim_start_matches("\\\\?\\")
+            .into();
+        let cargo_metadata = self
+            .wdk_build_provider
+            .get_cargo_metadata_at_path(&working_dir_path_trimmed)?;
+        Ok(cargo_metadata)
+    }
+
+    fn build_and_package(
+        &self,
+        working_dir: &PathBuf,
```

Comment:
Use `&Path` instead of `&PathBuf` anywhere you need a borrowed path. A paramter of type `&Path` is more flexible as it can accept both `&Path` and `&PathBuf` as arguments.